src/kbmod/reprojection_utils.py

```diff
@@ -38,28 +38,79 @@ def correct_parallax(coord, obstime, point_on_earth, guess_distance):
     # the object has an unknown distance from earth
     los_earth_obj = coord.transform_to(GCRS(obstime=obstime, obsgeoloc=loc))
 
-    cost = lambda d: np.abs(
-        guess_distance
-        - GCRS(ra=los_earth_obj.ra, dec=los_earth_obj.dec, distance=d * u.AU, obstime=obstime, obsgeoloc=loc)
+    cost = lambda geocentric_distance: np.abs(
+        heliocentric_distance
+        - GCRS(
+            ra=los_earth_obj.ra,
+            dec=los_earth_obj.dec,
+            distance=geocentric_distance * u.AU,
+            obstime=obstime,
+            obsgeoloc=loc,
+        )
         .transform_to(ICRS())
         .distance.to(u.AU)
         .value
     )
 
     fit = minimize(
         cost,
-        (guess_distance,),
+        (heliocentric_distance,),
     )
 
-    answer = GCRS(
-        ra=los_earth_obj.ra, dec=los_earth_obj.dec, distance=fit.x[0] * u.AU, obstime=obstime, obsgeoloc=loc
+    answer = SkyCoord(
+        ra=los_earth_obj.ra,
+        dec=los_earth_obj.dec,
+        distance=fit.x[0] * u.AU,
+        obstime=obstime,
+        obsgeoloc=loc,
+        frame="gcrs",
     ).transform_to(ICRS())
 
-    return answer
+    return answer, fit.x[0]
+
+
+def invert_correct_parallax(coord, obstime, point_on_earth, geocentric_distance, heliocentric_distance):
+    """Calculate the original ICRS coordinates of a point in EBD space, i.e. a result from `correct_parallax`.
+
+    Attributes
+    ----------
+    coord : `astropy.coordinate.SkyCoord`
+        The coordinate to be corrected for.
+    obstime : `astropy.time.Time` or `string`
+        The observation time.
+    point_on_earth : `astropy.coordinate.EarthLocation`
+        The location on Earth of the observation.
+    geocentric_distance : `float`
+        The distance from Earth to the object (generally a result from `correct_parallax`).
+    heliocentric_distance : `float`
+        The distance from the solar system barycenter to the object (generally an input for `correct_parallax`).
+
+    Returns
+    ----------
+    An `astropy.coordinate.SkyCoord` containing the ra and dec of the point in ICRS.
+
+    References
+    ----------
+    .. [1] `Jupyter Notebook <https://github.com/maxwest-uw/notebooks/blob/main/uncorrecting_parallax.ipynb>`_
+    """
+    loc = (
+        point_on_earth.x.to(u.m).value,
+        point_on_earth.y.to(u.m).value,
+        point_on_earth.z.to(u.m).value,
+    ) * u.m
```

Comment:
Do you need to do the `.value` and `* u.m`? Since you are using `.to(u.m)` the values might already have the correct units appended.